legend-engine-language-pure-dsl-service/src/main/java/org/finos/legend/engine/protocol/pure/v1/model/packageableElement/service/PureInlineExecution.java

```diff
@@ -0,0 +1,19 @@
+// Copyright 2022 Goldman Sachs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package org.finos.legend.engine.protocol.pure.v1.model.packageableElement.service;
+
+public class PureInlineExecution extends PureExecution
```

Comment:
I am not sure if we should add another type of PureExecution calling it PureInlineExecution, since we ultimately want to migrate all PureSingleExecution / PureMultiExecution to this.
I feel it would be ideal if users don't need to specify anything - Inline/Single/Multi and if users have added ExecutionEnvironment in grammar we infer it but not sure if that would be good for first pass

But maybe we can reuse Single & Multi itself and make adjustments to that so that its easy for us to make changes in composer and auto migrate services to this format in future

What do you think?

Reply:
I think so as well, it would be much better